client/client-lib/src/mint/backup.rs

```diff
@@ -259,18 +259,20 @@ impl MintClient {
         backup: PlaintextEcashBackup,
         gap_limit: usize,
     ) -> Result<Cancellable<EcashRecoveryFinalState>> {
-        let end_epoch = match self.context.api.fetch_last_epoch().await {
+        let epoch_count = match self.context.api.fetch_epoch_count().await {
             Ok(v) => v,
             Err(e) => {
                 return Err(e.into());
             }
         };
-        let epoch_range = backup.epoch..=end_epoch;
+        let start_epoch = if backup.epoch_count > 0 {
+            backup.epoch_count - 1
```

Comment:
Handle underflow with `saturating_sub`

Reply:
Thanks for this tip. I've changed that now.